### PR TITLE
[WIP] AZP/PKG: Publish Snapshot packages to Azure Artifacts.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Publish snapshot packages to Azure Artifacts

## Why ?
1. To be easier to test fixes, before release.
2. To easily distribute fixes on different environments.

We probably need to create a public feed in AZP (cc @avildema): 
image

https://docs.microsoft.com/en-us/azure/devops/artifacts/concepts/feeds?view=azure-devops

@yosefe  WDYT?

### CODE REVIEW

**Comment by yosefe** [comment]
@petro-rudenko is it like a "nightly build" where we could have latest builds of UCX packages?

**Comment by petro-rudenko** [comment]
Correct on every merge commit to master build and upload say `ucx-master-29-01-2021-ubuntu18.04-mofed5.0-cuda11.1.deb` to Azure artifactory.

**Comment by yosefe** [comment]
> Correct on every merge commit to master build and upload say `ucx-master-29-01-2021-ubuntu18.04-mofed5.0-cuda11.1.deb` to Azure artifactory.

Sounds good
I guess we'd to upload all binary packages (rpm/deb),src.rpm,.tgz, and .jar?
Do you know if there is storage limit/expiration time on this artifactory?

**Comment by petro-rudenko** [comment]
Jar we already uploading to maven central snapshot repository on every master commit.

`Azure Artifacts includes a free usage tier of 2 GB.`  Our packages are small:

image

We could also retain for a month or so, after remove old packages.

**Comment by petro-rudenko** [comment]
The job is pretty simple. It'll do the same release job now does, but upload not to github release page, but to Azure artifact

File: buildlib/azure-pipelines-pr.yml [hunk 1]
```diff
@@ -225,3 +225,7 @@ stages:
     dependsOn: [Codestyle]
     jobs:
     - template: io_demo/io-demo.yml
+
+  - stage: Release
+    jobs:
+      - template: az-distro-release.yml
```
Comment (avildema) [inline buildlib/azure-pipelines-pr.yml:231]: add space line to the end of file

**Pull request closed without merging**